tests/Benchmark/SimpleSetupBench.php

```diff
@@ -49,6 +50,10 @@ public function setUp(): void
         $this->store = new DoctrineDbalStore(
             $connection,
             DefaultEventSerializer::createFromPaths([__DIR__ . '/BasicImplementation/Events']),
+            DefaultHeadersSerializer::createFromPaths([
```

Comment:
Hm, I'm not sure about that. Due to that it's more masked for the user that this is happening and if he implements his own version of the `HeadersSerializer` he needs to know this internal logic in our `DefaultHeadersSerializer`.

But if we we do this then i think we should keep it as auto discover due two reasons:
1. We cannot forget to add headers then there
2. No need for extra logic

Reply:
Then we are duplicating the naming code 🤔 
So i think, if we move it, it should land in `AttributeMessageHeaderRegistryFactory` but still feels weird.
```php
        $classes = (new ClassFinder())->findClassNames($paths);
        $classes = array_merge([AggregateHeader::class, ...], $classes);
```

Reply:
> Then we are duplicating the naming code 🤔

I do not understand that.

For me, `MessageHeaderRegistry` would be the more correct place. I wouldn't put it any further down.

Reply:
If you mean the attributes. I would remove these from the internal headers because the attributes are only used for configuration. And since we define the internal headers differently, they have no meaning there. The attribute is for user space.